src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -348,6 +348,10 @@ private ImmutableArray<BoundExpression> BuildArgumentsForDynamicInvocation(Analy
                         {
                             builder[i] = ((OutVarLocalPendingInference)argument).FailInference(this, diagnostics);
                         }
+                        if (argument.Kind == BoundKind.DefaultPendingInference)
+                        {
+                            builder[i] = ((BoundDefaultPendingInference)argument).FailInference(this, diagnostics);
+                        }
 
```

Comment:
Here, too, the problem is an expression without a type, not this particular expression form.